src/transformers/models/conditional_detr/configuration_conditional_detr.py

```diff
@@ -93,11 +93,12 @@ class ConditionalDetrConfig(PretrainedConfig):
         position_embedding_type (`str`, *optional*, defaults to `"sine"`):
             Type of position embeddings to be used on top of the image features. One of `"sine"` or `"learned"`.
         backbone (`str`, *optional*, defaults to `"resnet50"`):
-            Name of convolutional backbone to use in case `use_timm_backbone` = `True`. Supports any convolutional
-            backbone from the timm package. For a list of all available models, see [this
-            page](https://rwightman.github.io/pytorch-image-models/#load-a-pretrained-model).
+            Name of backbone to use when `backbone_config` is `None`. If `use_pretrained_backbone` is `True`, this
+            will load the corresponding pretrained weights from the timm or transformers library. If `use_pretrained_backbone`
+            is `False`, this loads the backbone's config and uses that to initialize the backbone with random weights.
```

Comment:
both doc changes imply that `use_timm_backbone` is no longer a requirement for this. Just wondering if this works when `use_timm_backbone=False`

Reply:
Yes! That's the main benefit of this PR. Previously, only timm models could be loaded as pretrained weights but the previous constraint is just a hangover of how detr models were implemented before we properly had backbone and how Autobackbone is currently used. 

However, not all of the arguments are fully compatible yet. This PR enables loading in the pretrained backbones, but most models will only use `AutoBackbone.from_config` or timm's `create_model` e.g. [here](https://github.com/huggingface/transformers/blob/45b1dfa342f2ec00634c31f635bded6e346c98d2/src/transformers/models/conditional_detr/modeling_conditional_detr.py#L366) and [here](https://github.com/huggingface/transformers/blob/45b1dfa342f2ec00634c31f635bded6e346c98d2/src/transformers/models/conditional_detr/modeling_conditional_detr.py#L352) for conditional detr. 

So, `use_pretrained_backbone=True`, `use_timm_backbone=False` are valid, but won't work until the follow up PR to use `load_backbone`. Which admiteddly is not great behaviour. I can add a validation check for this combo until it's fully enabled?

Reply:
yep sounds good!

Reply:
Done!  https://github.com/huggingface/transformers/pull/28214/commits/5e565e32a41f7162fbc96d41441faaf4127f0667